blaze-client/src/main/scala/org/http4s/blaze/client/BlazeClient.scala

```diff
@@ -85,6 +86,7 @@ object BlazeClient {
           borrow.use { next =>
             val res: F[Resource[F, Response[F]]] = next.connection
               .runRequest(req)
+              .adaptError { case EOF => new ResponseException(req) }
```

Comment:
Other exceptions can be raised by Blaze, and they're worth contemplating.  We explicitly test that a `ConnectException` percolates up.  That has request info in its message, but does not return the request for further introspection.  Should we wrap that too?  Discuss.